src/Image/Svg.php

```diff
@@ -2824,6 +2824,93 @@ function svgWriteString($content)
 		$this->svg_string .= $content;
 	}
 
+	/**
+	 * SVGs made with Adobe Illustrator use a style tag and classes instead of inline styles
+	 * See: https://github.com/mpdf/mpdf/issues/450
+	 *
+	 * This function brutally copies the styles inline
+	 *  ( Currently only looks for classes as a selector )
+	 *
+	 * @param string $data svg contents
+	 * @return string svg contents
+	 * @author Antonio Norman - softcodex.ch
+	 */
+	function mergeStyles($data)
+	{
+		try {
+			$xml = new \DOMDocument();
+			$xml->loadXML($data);
+
+
+			// Check it's an SVG
+			$svgNode = $xml->getElementsByTagName('svg');
+			if ($svgNode->length === 0) {
+				return $data;
+			}
+
+			// Find the style node
+			$styles = [];
+			/** @var $styleNode \DOMNode */
+			foreach ($svgNode->item(0)->getElementsByTagName('style') as $styleNode) {
+
+				preg_match_all('/(\.[^{]+)\s*\{\s*([^}]+)\s*}/m', $styleNode->nodeValue, $matches, PREG_SET_ORDER);
+				foreach ($matches as $cssBlock) {
+					$css = preg_replace('/\s{2,}/', ' ', $cssBlock[2]); // Clean spaces or new lines
+					$selector = trim($cssBlock[1]);
+
+					$styles[$selector] = isset($styles[$cssBlock[1]]) ?
+						$styles[$selector] . ' ' . $css : // Append if the selector is already defined
+						$css;
+				}
+			}
+
+			if (empty($styles)) {
+				return $data;
+			}
+
+			// Recursively loop the nodes inserting the styles inline
+			$setStylesInline = function (\DOMNode $xml) use ($styles, &$setStylesInline) {
+				// Apply the styles to the elements
+				foreach ($xml->childNodes as $node) {
+
+					if ($node instanceof \DOMElement) {
+
+						// Check the node has the a class with a style
+						if ($node->hasAttribute('class')) {
+
+							// Allow for class=" class1  class2 "
+							$classes = explode(' ', $node->getAttribute('class'));
+
+							foreach ($classes as $class) {
+
+								$class = '.' . trim($class);
+								if (!empty($class) && isset($styles[$class])) {
+
+									$style = $node->hasAttribute('style') ?
+										$styles[$class] . ' ' . $node->getAttribute('style') :
+										$styles[$class];
+
+									$node->setAttribute('style', $style);
+								}
+							}
+						}
+
+						if ($node->hasChildNodes()) {
+							$setStylesInline($node);
+						}
+					}
+				}
+			};
+
+			$setStylesInline($xml);
+
+		} catch (\Exception $e) {
```

Comment:
Where exactly can the exception be thrown? Can't you surround only relevant code portions to reduce indentation?

Also, catching \Exception directly is not recommended, DOM methods throw an \DOMException, so you should catch that only to prevent catching unrelated exceptions, especially when you throw the exception details away in the catch block.

Reply:
I thought DOMDocument was throwing an Exception, in fact it's issuing a Error (for malformed XML from another test) that PHPUnit was converting to an Exception.
I'll silence the Error as it's not much help in a PDF generation. `$xml->loadXML($data, LIBXML_NOERROR )`, the other option is to register an error handler to convert the Error to Exception. 

Surrounding the whole block I don't think is too bad as it's an "atomic" operation any Exception means something has gone wrong during the transformation. I prefer a SVG with missing colours to no SVG.  (Anyway I'll remove the try & catch)

I agree the indentation is terrible, I'll invert the ifs in the loop.